src/LoadingStoreValue.ts

```diff
@@ -0,0 +1,103 @@
+import LoadingStoreCollection from './LoadingStoreCollection'
+import Resource from './interfaces/Resource'
+import { QueryablePromise, wrapPromise } from './QueryablePromise'
+
+/**
+ * Creates a placeholder for an entity which has not yet finished loading from the API.
+ * Such a LoadingStoreValue can safely be used in Vue components, since it will render as an empty
+ * string and Vue's reactivity system will replace it with the real data once that is available.
+ *
+ * Accessing nested functions in a LoadingStoreValue yields another LoadingStoreValue:
+ * new LoadingStoreValue(...).author().organization() // gives another LoadingStoreValue
+ *
+ * Using a LoadingStoreValue or a property of a LoadingStoreValue in a view renders to empty strings:
+ * let user = new LoadingStoreValue(...)
+ * 'The "' + user + '" is called "' + user.name + '"' // gives 'The "" is called ""'
+ *
+ * @param entityLoaded a Promise that resolves to a StoreValue when the entity has finished
+ *                     loading from the API
+ * @param absoluteSelf optional fully qualified URI of the entity being loaded, if available. If passed, the
+ *                     returned LoadingStoreValue will return it in calls to .self and ._meta.self
+ */
+class LoadingStoreValue implements Resource {
+  public _meta: {
+    self: string | null,
+    load: QueryablePromise<Resource>
+    loading: boolean
+  }
+
+  private loadResourceSafely: Promise<Resource>
+
+  constructor (entityLoaded: Promise<Resource>, absoluteSelf: string | null = null) {
+    this._meta = {
+      self: absoluteSelf,
+      load: wrapPromise(entityLoaded),
+      loading: true
+    }
+
+    // safe load: if API call fails, suppress errors and present this Proxy again to the chain
```

Comment:
The user of the package still knows because the original load promise is returned back (e.g. via patch(), post(), etc.).

I had a situation where my tests failed randomly and it took me quite some time to track it down to this. Unfortunately I cannot reproduce now anymore, so removing `.catch(() => this)` right now would still pass all tests.

What happens is that every `.then()` on a promise creates a new promise. If the original promise throws an error, every promise created by `.then()` potentially throws one, too.

See example here:
https://jsfiddle.net/r457jqm2/

If you run this example you should see one console output "Crap", but also 3 unhandled promise errors. This might be a nuisance only in browser environment. However node doesn't like this. In best case, we get a "Unhandled promise rejection" warning. In worst case, tests fail that are completely unrelated to this code because they get failing promises from previous test runs.

Reply:
Removed with this commit:
https://github.com/ecamp/hal-json-vuex/pull/9/commits/8365df6908aefd97290c55b413ade227193aee01